.github/actions/setup_environment/action.yml

```diff
@@ -0,0 +1,42 @@
+name: 'Setup environment for demisto-sdk'
+description: 'Setup environment for every demisto-sdk workflow job'
+author: 'Demisto-SDK'
+
+inputs:
+  python-version:
+    required: true
+    type: string
+    description: "The python version"
+  node-version:
+    required: false
+    type: string
+    default: "16"
+    description: "The node version to install"
+
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Install SDK dependencies with poetry
+      run: |
+        sudo curl -sSL https://install.python-poetry.org | python3 -
+        poetry --version
+        poetry check --lock
```

Comment:
I think it's worth moving `poetry check` to a different step, so it the poetry.lock and pyproject.toml are inconsistent it will fail on this step.
In addition, we can use install poetry action instead. I believe they take care on cache and so on

Reply:
done